dotnet/src/SemanticKernel/Planning/IPlan.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+/// <summary>
+/// Interface for a plan executable by the kernel
+/// </summary>
+public interface IPlan
+{
+    /// <summary>
+    /// State of the plan
+    /// </summary>
+    ContextVariables State { get; }
+
+    /// <summary>
+    /// Root step of the plan
+    /// </summary>
+    PlanStep Root { get; }
+
+    /// <summary>
+    /// Run the next step of the plan
+    /// </summary>
+    /// <param name="kernel">Kernel instance to use</param>
+    /// <param name="variables">Variables to use</param>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>The updated plan</returns>
+    Task<IPlan> RunNextStepAsync(
```

Comment:
In the dev tools we want to be able to determine the out of the step, detect errors and also collect some metadata.

1. How would a caller determine the output of the current step? I think the caller needs a reference to the step that just executed, how do you get that? It would be good to have some documentation explaining this behaviour.
2. How would the caller determine that the step execution failed?
3. Is there any metadata collected for each step execution e.g., token counts, duration, ...

Reply:
Great callouts. Will likely add to this interface like the `SkillPlan` has today. But for now was just tyring to make the min interfaces before starting the next PRs.